### PR TITLE
CNTools 9.0.0

### PR DESCRIPTION
**Pull request by Scitz0**
Koios integration into CNTools for utxo and pool info.

### CODE REVIEW

File: files/grest/rpc/address/address_info.sql [hunk 1]
```diff
@@ -2,43 +2,44 @@ DROP FUNCTION IF EXISTS grest.address_info (text);
 
 CREATE FUNCTION grest.address_info (_address text DEFAULT NULL)
   RETURNS TABLE (
-    balance numeric,
```
Comment (DamjanOstrelic) [inline files/grest/rpc/address/address_info.sql:5]: Any specific reason we are switching this field to string?

**Review by rdlrt — APPROVED** [review]
Brilliant overall! Few considerations:
- Add logging for every curl request made to Koios by default to cntools-history.log
- Knowing previous community requests, rather than relying on unset of KOIOS_API variable OR editing it to invalid value, would be good to have a ENABLE_KOIOS flag instead (defaulting to 'Y") in env
- For ` >> ADVANCED >> CHAIN-ANALYSIS >> POOL QUERIES >> METADATA HASH` , not sure if we need to show the script before execution, could possibly reduce the prompt?

**Comment by Scitz0** [comment]
> Brilliant overall! Few considerations:
> 
> * Add logging for every curl request made to Koios by default to cntools-history.log
👍 
> * Knowing previous community requests, rather than relying on unset of KOIOS_API variable OR editing it to invalid value, would be good to have a ENABLE_KOIOS flag instead (defaulting to 'Y") in env
👍 **sounds good as well.** 
> * For ` >> ADVANCED >> CHAIN-ANALYSIS >> POOL QUERIES >> METADATA HASH` , not sure if we need to show the script before execution, could possibly reduce the prompt?
**Could print to less as we do for changelog, in case of big script.**

**Comment by Scitz0** [comment]
Updates added according to notes.

**Pull request merged**